sparseSpACE/HP_Optimization.py

```diff
@@ -0,0 +1,588 @@
+		elif(name == "gaussian_quantiles"):
+			dataset = deml.datasets.make_gaussian_quantiles(n_samples=self.samples, n_features=self.dimension, n_classes=self.labels, random_state=1)
+		elif(name == "digits"):
+			dataset = deml.datasets.load_digits(return_X_y=True) # hint: try only with max_level <= 3
+		elif(name == "iris"):
+			dataset = deml.datasets.load_iris(return_X_y=True)
+		elif(name == "breast_cancer"):
+			dataset = deml.datasets.load_breast_cancer(return_X_y=True) # hint: try only with max_level <= 4
+		elif(name == "wine"):
+			dataset = deml.datasets.load_wine(return_X_y=True)
+		elif(name != "moons"):
+			print("This name for the data set is unknown, using moons.\nPlease check that the name is not capitalized, words are separated by _, and that it is written correctly")
+		data = deml.DataSet(dataset, name="data_"+name)
+		return data
+
+	def pea_classification(self, params):
+		if(len(params)<4):
+			print("too little parameters for pea_classification. Returning 0.0")
+			return 0.0
+		params = [params[0], int(params[1]), int(params[2]), int(params[3])]
+		cur_data = self.data.copy()
+		#should implement a smoother way to put in the data set
+		classification = deml.Classification(cur_data, split_percentage=0.8, split_evenly=True, shuffle_data=False)
+		classification.perform_classification(masslumping=params[1], lambd=10**(-params[0]), minimum_level=params[2], maximum_level=self.max_lv, one_vs_others=params[3], print_metrics=False)
+		evaluation = classification.evaluate()
+		#wenn Zeit mit reingerechnet werden soll o.ä. in dieser Funktion
+		return evaluation["Percentage correct"]
+
+	def pea_classification_dimension_wise(self, params):
+		if(len(params)<4):
+			print("too little parameters for pea_classification. Returning 0.0")
+			return 0.0
+		#lambd, massl, min_lv, one_vs_others / error calc, margin(float 0-1), rebalancing (bool), use_relative_surplus (bool), max_evaluations (int)
+		params = [float(params[0]), int(params[1]), int(params[2]), int(params[3]), float(params[4]), int(params[5]), int(params[6]), int(params[7])]
+		cur_data = self.data.copy()
+		error_calculator=sparseSpACE.ErrorCalculator.ErrorCalculatorSingleDimMisclassificationGlobal()
+		ec = None
+		ovo = False
+		if(params[3] == 2):
+			ec = error_calculator
+			ovo = True
+		elif(params[3] == 1):
+			ovo = True
+		classification = deml.Classification(cur_data, split_percentage=0.8, split_evenly=True, shuffle_data=False)
+		classification.perform_classification_dimension_wise(masslumping=params[1], lambd=10**(-params[0]), minimum_level=params[2], maximum_level=self.max_lv, one_vs_others=ovo, error_calculator = ec, margin = params[4], rebalancing = params[5], use_relative_surplus = params[6], max_evaluations = params[7], print_metrics=False)
+		evaluation = classification.evaluate()
+		#wenn Zeit mit reingerechnet werden soll o.ä. in dieser Funktion
+		return evaluation["Percentage correct"]
```

Comment:
Etwas mehr Kommentare und vor allem korrekte Doku mit der Beschreibung der Inputparameter (siehe den Dokumentationsstyle z.B. in DEMachineLearning.py -> """ Beschreibung  Params""")